discovery_engine_core/web-api2/src/elastic.rs

```diff
@@ -140,33 +137,60 @@ impl ElasticSearchClient {
         Ok(())
     }
 
+    pub async fn bulk_insert_documents(
+        &self,
+        documents: &[(DocumentId, ElasticDocument)],
+    ) -> Result<(), BulkInsertionError> {
+        let response = self
+            .bulk_request(documents.iter().flat_map(|(document_id, document)| {
+                [
+                    serde_json::to_value(BulkInstruction::Index { id: document_id })
+                        .map_err(Into::into),
+                    serde_json::to_value(document).map_err(Into::into),
+                ]
+            }))
+            .await?;
+
+        if !response.errors {
+            Ok(())
+        } else {
+            let failed_documents = response.items
+                .into_iter()
+                .filter_map(|mut response| {
+                    if let Some(response) = response.remove("index") {
+                        if is_success_status(response.status, false) {
```

Comment:
Shouldn't this be the opposite check?
```suggestion
                        if !is_success_status(response.status, false) {
```